internal/venice-common/src/main/java/com/linkedin/venice/meta/VersionImpl.java

```diff
@@ -31,23 +31,40 @@ public class VersionImpl implements Version {
    * Currently used in tests only.
    */
   @Deprecated
-  public VersionImpl(String storeName, int number) {
+  public VersionImpl(String storeName, int number, String realTimeTopicName) {
```

Comment:
Question: shall we keep the old interface `VersionImpl(String storeName, int number)` and add a new one for explicit setting the realtime topic name? It could reduce lots of changes to the existing testing etc. what do you think? is there any downsides of keeping the old one?

Reply:
Yes, it was a pain to update all the references.
The only reason I added it in all the existing constructors because it would be a required field in Version and forgetting to explicitly set it will leads to discrepancy and bugs. @FelixGV do you have any thoughts?

Reply:
IMHO it is fine to add this param to all constructors since, as Arjun said, if we leave a constructor without this param and let it fill in the new field with today's behavior (i.e. `Version::composeRealTimeTopic`), then we'll probably end up with bugs...

Most likely, in the end state, we need to completely delete `Version::composeRealTimeTopic`, so making use of it in one of the default constructors here goes against that...

Reply:
I see. It makes sense to me as we will delete its usage eventually.